System/RTEError.cpp

```diff
@@ -7,73 +7,403 @@
-			bool abortSaveMade = false;
-			if (g_ActivityMan.GetActivityAllowsSaving()) {
-				abortSaveMade = g_ActivityMan.SaveCurrentGame("AbortSave");
+			// This typically contains the absolute path to the file on whatever machine this was compiled on, so in that case get only the file name.
+			std::filesystem::path filePath = srcLocation.file_name();
+			std::string fileName = (filePath.has_root_name() || filePath.has_root_directory()) ? filePath.filename().generic_string() : srcLocation.file_name();
+
+			std::string lineNum = std::to_string(srcLocation.line());
+			std::string funcName = srcLocation.function_name();
+
+			std::string abortMessage = "Runtime Error in file '" + fileName + "', line " + lineNum + ",\nin function '" + funcName + "'\nbecause:\n\n" + description + "\n";
+
+			if (DumpAbortSave()) {
+				abortMessage += "\nThe game has saved to 'AbortSave'.";
+			}
+			if (DumpAbortScreen()) {
+				abortMessage += "\nThe last frame has been dumped to 'AbortScreen.png'.";
+			}
+
+			g_ConsoleMan.PrintString(abortMessage);
+
+			std::string callstack = "";
+
+#ifdef _WIN32
+			RTEStackTrace stackTrace;
+			callstack += ("\n\n" + stackTrace.GetCallStackAsString());
+#endif
+
+			std::string consoleSaveMsg;
+			if (!callstack.empty()) {
+				g_ConsoleMan.PrintString(callstack);
+				consoleSaveMsg = "\nThe console and callstack have been dumped to 'AbortLog.txt'.";
+			} else {
+				consoleSaveMsg = "\nThe console has been dumped to 'AbortLog.txt'.";
 			}
 
-			if (BITMAP *backbuffer = g_FrameMan.GetBackBuffer32(); backbuffer) {
-				// Have to convert the 32bpp backbuffer to 24bpp otherwise the saved file is blank for reasons that don't matter.
-				BITMAP *abortScreenBuffer = create_bitmap_ex(24, backbuffer->w, backbuffer->h);
-				blit(backbuffer, abortScreenBuffer, 0, 0, 0, 0, backbuffer->w, backbuffer->h);
-				save_png("AbortScreen.png", abortScreenBuffer, nullptr);
-				destroy_bitmap(abortScreenBuffer);
+			if (g_ConsoleMan.SaveAllText("AbortLog.txt")) {
+				abortMessage += consoleSaveMsg;
 			}
+			System::PrintToCLI(abortMessage);
 
 			// Ditch the video mode so the message box appears without problems.
 			if (g_WindowMan.GetWindow()) {
 				SDL_SetWindowFullscreen(g_WindowMan.GetWindow(), 0);
 			}
 
-			// This typically gets passed __FILE__ which contains the full path to the file from whatever machine this was compiled on, so in that case get only the file name.
-			std::filesystem::path filePath = file;
-			std::string fileName = (filePath.has_root_name() || filePath.has_root_directory()) ? filePath.filename().generic_string() : file;
+			if (ShowAbortMessageBox(abortMessage)) {
+				// Enable restarting in release builds only.
+				// Once this exits the debugger is detached and while there does seem to be a way to programatically re-attach it to the new instance (at least in Windows), it is so incredibly ass and I cannot even begin to can.
```

Comment:
You can attach to process with vs, and there's also reattach to process, but I'm guessing that won't be helpful here. Still, I agree that it's better to avoid the potential confusion.

Reply:
`programatically` is the keyword here.